Create API endpoint for proximity analysis

yaroluchko opened the pull request:
This pull request: 

- creates the API endpoint for the proximity analysis posting
- creates an API endpoint for viewing all the proximity analyses
- adds the urls for the two API endpoints above
- adds a manual test in the proximity_test.py file

Watermelanie committed:
created an API endpoint for posting proximity analysis

Co-Authored-By: yaroluchko <85574367+yaroluchko@users.noreply.github.com>

joshfeli committed:
Resolved `__hash__` error

joshfeli committed:
Merge branch 'main' into create_API_endpoint_for_proximityAnalysis

yaroluchko committed:
Merge branch 'main' into create_API_endpoint_for_proximityAnalysis

yaroluchko committed:
Merge branch 'create_API_endpoint_for_proximityAnalysis' of https://github.com/dhmit/gender_analysis_web into create_API_endpoint_for_proximityAnalysis

yaroluchko committed:
Merge branch 'main' into create_API_endpoint_for_proximityAnalysis

yaroluchko committed:
Proximity Analysis API endpoint

This creates the Proximity Analysis API endpoint and a way to see all the analyses

yaroluchko committed:
A fix to make the key a string instead of Gender

yaroluchko committed:
Create proximity_test.py

A manual test for proximity posting

yaroluchko committed:
Update views.py

Fix unncessary result calculation if already done before

yaroluchko committed:
Update views.py

joshfeli commented:
Just to note: The `proximity_frontend` branch has this copied and pasted to test the endpoint with the button in place, and all data passes through successfully! We were able to at least get the console to print out the `results` object, but this will be replaced with a display function.

MBJean commented on backend/app/views.py:
```diff
@@ -243,6 +244,43 @@ def get_corpus(request, corpus_id):
+            'word_window': word_window,
+            'results': results,
+        }
+
```
Ultimately, we may want to change this up so that you call a ProximityAnalysisManager or similar (like we do with the DocumentManager) when we create this new ProximityAnalysis, and have the ProximityAnalysisManager call the run_analysis function. That is: since a ProximityAnalysis necessarily relies on the run_analysis function, we could use the ProximityAnalysis itself (through a Manager) to ensure that run_analysis is run when a new instance is created. That would more tightly couple associated functionality. That said, since this is working and is well tested, we can leave it be for now and come back to it later.

MBJean commented on backend/app/analysis/proximity_test.py:
```diff
@@ -0,0 +1,18 @@
+    factory = APIRequestFactory()
+    request = factory.post('api/all_proximity', {'word_window': '2', 'corpus_id': c1.id})
+    add_proximity_analysis(request)
+
```
Was this test used simply to manually ensure that the full ProximityAnalysis creation occurred as expected? Would it be possible to add something to our larger test suite so we can check that the model instance is saved as expected?

yaroluchko replied:
Yea, this was just a manual test. I will try and add a more comprehensive test.

yaroluchko replied:
I've just added a test to the tests.py file that makes sure the post request is successful

MBJean reviewed (commented):
This generally looks great! Thanks for your updates. I've left a few comments below. I'd be interested in seeing a more comprehensive test, if possible.

yaroluchko committed:
Test Case for proximity analysis object creation

yaroluchko committed:
Delete proximity_test.py

joshfeli committed:
Altered import statement

`frequency.py` also has a `run_analysis` function; this change will prevent aliasing errors should the other team import the `frequency` counterpart into `views.py`.

joshfeli committed:
Squashed commit of the following:

commit 0fdfcc6ada81e5f5b1e142827bc50d30e2148b0a
Merge: 1276da0 b10cfa9
Author: Peihua Huang <32581282+phuang00@users.noreply.github.com>
Date:   Fri Jul 23 15:48:02 2021 -0400

    Merge pull request #25 from dhmit/transfer_frequency_module_PR1

    Transfer frequency module pr1

commit 1276da064e4fd5dcc9da858aa78d852524c54b6d
Merge: 657b20f b5a9482
Author: YifanWang0 <47550739+YifanWang0@users.noreply.github.com>
Date:   Fri Jul 23 15:38:12 2021 -0400

    Merge pull request #54 from dhmit/transfer_frequency_model

    Added FrequencyAnalysis model

commit b5a948238525814c9086af246e3197f1b2f0016d
Author: Yifan Wang <yifan.wang0422@gmail.com>
Date:   Fri Jul 23 14:08:54 2021 -0400

    added FrequencyAnalysis model

commit b10cfa95e08ad4132ec3a3391649d85e8fa5c234
Author: Peihua Huang <nemesis2126@gmail.com>
Date:   Fri Jul 23 13:41:17 2021 -0400

    update run_analysis to use primary keys instead of labels and update result dictionary to key by gender object

commit 1196fc14726aeb228902877e2ea55311654e1321
Author: Peihua Huang <nemesis2126@gmail.com>
Date:   Fri Jul 23 13:36:54 2021 -0400

    Revert "update frequency result to key by gender primary key instead of label"

    This reverts commit 151a4bc066c66225e6f6c6b8cd8fcf9d673df7b6.

commit 151a4bc066c66225e6f6c6b8cd8fcf9d673df7b6
Author: Peihua Huang <nemesis2126@gmail.com>
Date:   Fri Jul 23 13:34:11 2021 -0400

    update frequency result to key by gender primary key instead of label

commit 657b20f92d417acfbc0cc633c21ce596da7208ea
Merge: 6619e68 3c6e5bf
Author: Joshua Feliciano <41080599+joshfeli@users.noreply.github.com>
Date:   Fri Jul 23 11:42:15 2021 -0400

    Merge pull request #52 from dhmit/remove_proximity_breakpoint

    Removed debugging breakpoint

commit 3c6e5bf9170190342a18605022fff1b8e8a38b91
Author: Joshua Feliciano <joshfeli613@gmail.com>
Date:   Thu Jul 22 17:50:15 2021 -0400

    Removed debugging breakpoint

commit 0684a02abd7ebee39e149b70671abefee63fa9c1
Author: Peihua Huang <nemesis2126@gmail.com>
Date:   Thu Jul 22 16:43:50 2021 -0400

    got frequency analysis working with a corpus

commit 2003ce64ffec878238c4f5e3d1c04c57a1135f5e
Author: Peihua Huang <nemesis2126@gmail.com>
Date:   Thu Jul 22 16:28:27 2021 -0400

    convert run_analysis to run_single_analysis, such that the function only takes in one document and update docstrings and added test

commit 0f817ba5ae500c05b4ecc0ce5ac7460c8e75271c
Merge: dc21391 6619e68
Author: Peihua Huang <nemesis2126@gmail.com>
Date:   Thu Jul 22 14:54:15 2021 -0400

    Merge branch 'main' into transfer_frequency_module_PR1

commit dc213913a32b5920863fce5fb5286ee1f2a071ca
Author: Yifan Wang <yifan.wang0422@gmail.com>
Date:   Wed Jul 7 16:44:55 2021 -0400

    fixed small mistakes according to pr comments

commit a852cd3db1a370eac384b04969cf3a27eaf42580
Author: Yifan Wang <yifan.wang0422@gmail.com>
Date:   Mon Jun 28 16:42:28 2021 -0400

    added docstrings

commit 32731f4ab3759730ded193bb115ea92e7f3baab5
Author: Yifan Wang <yifan.wang0422@gmail.com>
Date:   Mon Jun 28 16:01:57 2021 -0400

    fixed minor bug with _run_analysis function

commit 233e2e6602e99bc212fffab6442a72d60a83e8db
Author: Yifan Wang <yifan.wang0422@gmail.com>
Date:   Fri Jun 25 16:52:31 2021 -0400

    moved over run analysis but still need to fix bug with helper function
